services/assets.js

```diff
@@ -108,19 +108,57 @@ module.exports = class AssetsService {
    * @param  {String} value string to search by.
    * @return {Promise}
    */
-  static search({value, skip, limit} = {}) {
-    if (!value) {
-      return AssetsService.all(skip, limit);
-    } else {
-      return AssetModel
-        .find({
-          $text: {
-            $search: value
+  static search({value, limit, open, sortOrder, cursor} = {}) {
+    let assets = AssetModel.find({});
+
+    if (value) {
+      assets.merge({
+        $text: {
+          $search: value
+        }
+      });
+    }
+
+    if (open != null) {
+      if (open) {
+        assets.merge({
+          $or: [
+            {
+              closedAt: null
+            },
+            {
+              closedAt: {
+                $gt: Date.now()
+              }
+            }
+          ]
+        });
+      } else {
+        assets.merge({
+          closedAt: {
+            $lt: Date.now()
           }
-        })
-        .skip(skip)
-        .limit(limit);
+        });
+      }
     }
+
+    if (cursor) {
+      if (sortOrder === 'DESC') {
+        assets.merge({
+          created_at: {
+            $lt: cursor,
+          },
+        });
+      } else {
+        assets.merge({
+          created_at: {
+            $gt: cursor,
+          },
+        });
+      }
+    }
+
+    return assets.limit(limit);
```

Comment:
The result  should be sorted by `createdAt` depending on `sortOrder` otherwise the pagination is wrong.